org.scala-ide.sdt.core/src/org/scalaide/core/internal/project/BuildManagerFactory.scala

```diff
@@ -0,0 +1,59 @@
+package org.scalaide.core.internal.project
+
+import org.scalaide.core.internal.builder.zinc.EclipseSbtBuildManager
+import org.eclipse.core.runtime.IPath
+import org.eclipse.core.resources.IContainer
+import org.scalaide.core.internal.builder.zinc.SbtAwareBuildManager
+import scala.tools.nsc.Settings
+import org.scalaide.core.IScalaProject
+
+object BuildManagerFactory {
+  sealed trait BuildManagerFactorySelector
+  case object EclipseSbtBuildManagerFactorySelector extends BuildManagerFactorySelector
+  case object ProjectsDependentSbtBuildManagerFactorySelector extends BuildManagerFactorySelector
+  case object ScopesBuildManagerFactorySelector extends BuildManagerFactorySelector
+
+  def apply(typeOfFactory: BuildManagerFactorySelector): BuildManagerFactory = typeOfFactory match {
+    case EclipseSbtBuildManagerFactorySelector => new EclipseSbtBuildManagerFactory
+    case ProjectsDependentSbtBuildManagerFactorySelector => new ProjectsDependentSbtBuildManagerFactory
+    case ScopesBuildManagerFactorySelector => new ScopesBuildManagerFactory
+    case _ => ???
+  }
+
+  def apply(useScopeCompiler: Boolean): BuildManagerFactory =
+    if (useScopeCompiler) {
+      new ScopesBuildManagerFactory
+    } else {
+      new ProjectsDependentSbtBuildManagerFactory
+    }
+}
+
+sealed trait BuildManagerFactory {
+  /** @param owningProject
+   *  @param managerSettings settings from Eclipse IDE for Scala
+   *  @param addToClasspath a sequence of <code>IPath</code>s which are added to project's defaults
+   *  used by sbt compiler. Can be empty.
+   *  @param srcOutputs a sequence of source-output folders used by sbt compiler. If not set
+   *  all source-output for whole project are collected.
+   */
+  def buildManager(owningProject: IScalaProject, managerSettings: Settings,
```

Comment:
Is this factory really needed? In general I prefer simple things, so if there's no need for abstraction here (only one possible factory), I'd rather do away with all this boilerplate.